tools/jarcat/zip/writer_test.go

```diff
@@ -24,10 +24,64 @@ func TestAddZipFile(t *testing.T) {
 	assertExpected(t, "add_zip_file_test.zip", 0)
 }
 
+// getZipContents is a helper which returns a map of filename -> contents
+func getZipContents(zipfilePath string) (map[string][]byte, error) {
+	reader, err := zip.OpenReader(zipfilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer reader.Close()
+	res := make(map[string][]byte, len(reader.File))
+	for _, file := range reader.File {
+		if file.FileInfo().IsDir() {
+			continue
+		}
+		fReader, err := file.Open()
+		if err != nil {
+			return nil, err
+		}
+		defer fReader.Close()
+		contents := make([]byte, file.FileInfo().Size())
+		if _, err := io.ReadFull(fReader, contents); err != nil {
+			return nil, err
+		}
+		res[file.FileInfo().Name()] = contents
+	}
+	return res, nil
+}
+
+func TestAddZipFileConcatenatesSpecialFiles(t *testing.T) {
+	t.Run("Scala Akka reference.conf", func(t *testing.T) {
```

Comment:
I haven't added the other special cases because I don't know what a good fixture looks like; but it is with those in mind why I've called my test like that and have a subtest